src/enzo/Grid_LightBosonInitialize.C

```diff
@@ -0,0 +1,248 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+#include "ErrorExceptions.h"
+#include "macros_and_parameters.h"
+#include "typedefs.h"
+#include "global_data.h"
+#include "Fluxes.h"
+#include "GridList.h"
+#include "ExternalBoundary.h"
+#include "Grid.h"
+#include "CosmologyParameters.h"
+
+int QuantumGetUnits(float *DensityUnits, float *LengthUnits,
+		      float *TemperatureUnits, float *TimeUnits,
+		      float *VelocityUnits, FLOAT Time);
+
+int grid::LightBosonInitializeGrid(float CenterPosition)
+{  
+
+  NumberOfBaryonFields = 0;
+  FieldType[NumberOfBaryonFields++] = Density;
+  FieldType[NumberOfBaryonFields++] = Velocity1;
+  FieldType[NumberOfBaryonFields++] = Velocity2;
+  FieldType[NumberOfBaryonFields++] = Velocity3;
+  FieldType[NumberOfBaryonFields++] = TotalEnergy;
+  FieldType[NumberOfBaryonFields++] = RePsi;
+  FieldType[NumberOfBaryonFields++] = ImPsi;
+  FieldType[NumberOfBaryonFields++] = FDMDensity;
+
+
+  int size = 1, index, dim;
+  for (dim = 0; dim < GridRank; dim++)
+    size *= GridDimension[dim];
+
+   /* Return if this doesn't concern us. */
+  int UseParticles = 1;
+  FLOAT ParticleMeanDensity = .1;
+  static int CollapseTestParticleCount = 0;
+  int ParticleCount = 0;
+  if (ProcessorNumber != MyProcessorNumber) {
+    NumberOfParticles = (UseParticles > 0) ? 1 : 0;
+    for (dim = 0; dim < GridRank; dim++)
+      NumberOfParticles *= (GridEndIndex[dim] - GridStartIndex[dim] + 1);
+    return SUCCESS;
+  }
+
+  int field;
+  for (field = 0; field < NumberOfBaryonFields; field++)
+    if (BaryonField[field] == NULL)
+      BaryonField[field] = new float[size];
+
+
+  float TemperatureUnits = 1, DensityUnits = 1, LengthUnits = 1,
+    VelocityUnits = 1, TimeUnits = 1;
+
+  if (QuantumGetUnits(&DensityUnits, &LengthUnits, &TemperatureUnits,
+         &TimeUnits, &VelocityUnits, Time) == FAIL) {
+    ENZO_FAIL("Error in GetUnits.");
+  }
+
+  int RePsiNum, ImPsiNum, FDMDensNum;
+  RePsiNum = FindField(RePsi, FieldType, NumberOfBaryonFields);
+  ImPsiNum = FindField(ImPsi, FieldType, NumberOfBaryonFields);
+  FDMDensNum = FindField(FDMDensity, FieldType, NumberOfBaryonFields);
+
+
+  float coef = (5.9157166856e27*TimeUnits/pow(LengthUnits,2));
```

Comment:
realistically, should this be a double?  and should pow be POW?

Reply:
changed to double